### ISSUE TITLE
Implement country icons

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1094]
**Is your feature request related to a problem? Please describe.**
- mars-sim displays icons for all space agencies but it doesn't show icons for country

**Describe the solution you desire to see**
- Use these [icons](https://www.flaticon.com/packs/countrys-flags) for each country 

**Specifications  (please complete the following information):**
 - pre-3.7.0 build 8774 

**Additional context**
- Display country icons in Sponsor Tab Panel and on top panel of Person Unit Window and Settlement Unit Window.

**Comment by Urwumpe** [comment]
Would it be possible to include unicode glyphs next to the country name as smaller solution?

**Comment by mokun** [comment]
How do I load the unicode glyphs for each country's flag ?

I'm looking at https://en.m.wikipedia.org/wiki/Regional_indicator_symbol.

**Comment by mokun** [comment]
I suppose first I'll need to map a country name to its country code.
Next, I'll need a library to print its graphic emoji icon.

Found this [library](https://stackoverflow.com/questions/26231263/create-string-with-emoji-unicode-flag-countries).

Also [this](https://stackoverflow.com/questions/28065491/java-swing-countrycombobox-with-icons) for a country combobox.

**Comment by mokun** [comment]
I can't find any library pre-loaded with each country's unicode glyphs (flag symbol). 

If it's not available out there, we may have to roll in our own.

**Comment by Urwumpe** [comment]
This one looks like we could use it there, it should be able to map the country codes to the right glyph.

https://github.com/vdurmont/emoji-java

It is based on the same data presented here: https://apps.timwhitlock.info/emoji/tables/iso3166

**Comment by mokun** [comment]
> This one looks like we could use it there, it should be able to map the country codes to the right glyph. https://github.com/vdurmont/emoji-java

Yes I've added its maven artifact and try on it. It's no longer being updated. 

I've also added another more current package called `jemoji`  

I can get the country code in unicode already in String. 

However, how do I capture the picture as a `Icon` or `ImageIcon` ? 

I can get the string based on [here](
https://stackoverflow.com/questions/30494284/android-get-country-emoji-flag-using-locale).



Still trying to figure out.

**Comment by mokun** [comment]
> Searching for the country name should also work.

The svg flag icons in [here](https://commons.wikimedia.org/wiki/Category:Emoji_One_v1)  are from 1KB to over 200KB.

So given the 2-letter country code, we do need to see how to turn its predefined unicode into a SVG glyph.

**Comment by mokun** [comment]
@Urwumpe @bevans2000 

Guess I'll use the 4x3 icons [here](https://github.com/lipis/flag-icons) or [here](https://flagicons.lipis.dev/).

**Comment by Urwumpe** [comment]
Why do you need it as icon? Just use it as label. Its just a series of 16 bit unicode codepoints, consuming 4 bytes.  

The advantage of using Unicode is, that this is very easy to port to another client and the unicode has no additional memory  requirements (already loaded with the JRE/OS) compared to the (surely better looking) icons. Also the SVG icons may look small but require more memory when parsed. For just an "redundant" information (next to country name), I would prefer the unicode solution.

**Comment by mokun** [comment]
> Why do you need it as icon? Just use it as label. 

![image](https://github.com/mars-sim/mars-sim/assets/1168584/3e44ca05-79bf-4c61-bbee-06c89b065ee3)

See above, in case of Russia, The country code is `RU`, when I print the unicode string out in JLabel, it returns also as `RU` in Java, except in a different font. That's the so-called emoji glyph that's embedded in Java.

And that's it. There's no colorful flag symbol. And that's not exactly what I want to see yet. 

> Also the SVG icons may look small but require more memory when parsed.

Yes. It's probably a matter of preference. I simply want more UI elements in mars-sim to be less wordy and more graphical.

**Comment by bevans2000** [comment]
Can we close this issue now

**Comment by mokun** [comment]
not yet. I'd like to download and add country flag icon and display it somewhere.

**Comment by bevans2000** [comment]
Maybe we need to focus on closing down the open issues as there are too many open

**Comment by bevans2000** [comment]
Can we close this?

**Comment by mokun** [comment]
No they're not done yet.

**Comment by bevans2000** [comment]
We should focus on closing down these half finished issues. I'm focusing on those old ones assigned to me.

**Comment by bevans2000** [comment]
What's left to resolve in this one?

**Comment by mokun** [comment]
We don't have jpg, pnh or svg icons for each country flag yet.

**Comment by bevans2000** [comment]
Can we close this now? Seems to have been open for a long time

**Comment by Urwumpe** [comment]
I agree to dropping this from the current release. Give me an evening, I'll compile a list of which countries we have and which lack icons, this sounds like a job for me.  Maybe the solution isn't so far away. Also, since the number of countries and/or agencies is less than 500, would be a problem if we curate our own assets instead of looking for a new dependency?

**Comment by mokun** [comment]
@Urwumpe 

> Give me an evening, I'll compile a list of which countries we have and which lack icons, this sounds like a job for me.

These are the [countries[(https://github.com/mars-sim/mars-sim/wiki/Countries).


> See above, in case of Russia, The country code is RU, when I print the unicode string out in JLabel, it returns also as RU in Java, except in a different font. That's the so-called emoji glyph that's embedded in Java.
> And that's it. There's no colorful flag symbol. And that's not exactly what I want to see yet.

Unfortunately, printing those glyph as JLabel doesn't work.

**Comment by mokun** [comment]
>  proper country flags 

where at ?

**Comment by Urwumpe** [comment]
> > proper country flags
> 
> where at ?

For example here: [WikiProject: Flags](https://commons.wikimedia.org/wiki/Commons:WikiProject_Flags/flags)
